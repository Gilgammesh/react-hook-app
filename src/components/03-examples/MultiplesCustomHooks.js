import React from "react";
import { useCounter } from "../../hooks/useCounter";
import { useFetch } from "../../hooks/useFetch";
import "./styles.css";

const MultiplesCustomHooks = () => {
  const { counter, increment } = useCounter(1);

  const { data, loading, error } = useFetch(
    `https://www.breakingbadapi.com/api/quotes/${counter}`
  );
  const { quote_id, quote, author } = !!data && data[0];

  return (
    <div>
      <h1>Breaking Bad Quote</h1>
      <hr />

      {loading ? (
        <div className="alert alert-info text-center">Loading ...</div>
      ) : (
        <div className="blockquote text-right">
          <h4>{quote_id}</h4>
          <p className="mb-0">{quote}</p>
          <footer className="blockquote-footer">{author}</footer>
        </div>
      )}

      <button className="btn btn-success" onClick={() => increment(1)}>
        Siguiente quote
      </button>
    </div>
  );
};

export default MultiplesCustomHooks;
