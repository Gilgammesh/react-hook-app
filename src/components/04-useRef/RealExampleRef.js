import React, { useState } from "react";
import MultiplesCustomHooks from "../03-examples/MultiplesCustomHooks";
import "./styles.css";

const RealExampleRef = () => {
  const [show, setShow] = useState(false);

  return (
    <div>
      <h1>Real Example Ref</h1>
      <hr />
      {show && <MultiplesCustomHooks />}

      <button
        className="btn btn-info mt-4"
        onClick={() => {
          setShow(!show);
        }}
      >
        Show/Hide
      </button>
    </div>
  );
};

export default RealExampleRef;
