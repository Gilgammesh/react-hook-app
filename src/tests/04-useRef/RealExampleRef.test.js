import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import RealExampleRef from "../../components/04-useRef/RealExampleRef";
import MultiplesCustomHooks from "../../components/03-examples/MultiplesCustomHooks";

describe("Pruebas unitarias en el componente RealExampleRef.js", () => {
  /***********************************************************************************************************/
  test("se debe mostrar el componente <RealExampleRef /> correctamente", () => {
    const wrapper = shallow(<RealExampleRef />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(MultiplesCustomHooks).exists()).toBe(false);
  });
  /***********************************************************************************************************/
  test("se debe mostrar el componente <MultiplesCustomHooks /> correctamente", () => {
    const wrapper = shallow(<RealExampleRef />);
    wrapper.find("button").simulate("click");    
    expect(wrapper.find(MultiplesCustomHooks).exists()).toBe(true);
  });
});
