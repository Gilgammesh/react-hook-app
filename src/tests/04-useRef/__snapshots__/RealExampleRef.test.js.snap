// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias en el componente RealExampleRef.js se debe mostrar el componente <MultiplesCustomHooks /> correctamente 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-info mt-4"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;

exports[`Pruebas unitarias en el componente RealExampleRef.js se debe mostrar el componente <RealExampleRef /> correctamente 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-info mt-4"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
